packages/entity/src/EntityLoader.ts

```diff
@@ -199,6 +224,31 @@ export default class EntityLoader<
     fieldEqualityOperands: FieldEqualityCondition<TFields, N>[],
     querySelectionModifiers: QuerySelectionModifiers<TFields> = {}
   ): Promise<readonly Result<TEntity>[]> {
+    const invalidFieldValueErrors: Result<TEntity>[] = [];
```

Comment:
I'm not sure we want to return an array of failure results since each error doesn't correspond to an entity that failed to load, it corresponds to an invalid query field. I think of `Result<TEntity>` as containing an entity or the error saying why we couldn't load that specific entity.

It doesn't make sense to iterate over `invalidFieldValueErrors` the same way it'd make sense to iterate over an array of entities and failures to load those entities. Perhaps we should just construct an AggregateError here and throw, or throw an unaggregated error on the first invalid value we see.